### PR TITLE
pacific: mds/client: fail the request if the peer MDS doesn't support getvxattr op

### PR DESCRIPTION
**Pull request by lxbsz**
backport tracker: https://tracker.ceph.com/issues/57239 and https://tracker.ceph.com/issues/57363

---

backport of https://github.com/ceph/ceph/pull/47063 and https://github.com/ceph/ceph/pull/45307
parent tracker: https://tracker.ceph.com/issues/56529 and https://tracker.ceph.com/issues/54461

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by vshankar** [comment]
@lxbsz could you fix the yaml as done for quincy backport?

**Comment by lxbsz** [comment]
> @lxbsz could you fix the yaml as done for quincy backport?

@vshankar Already done last Friday!

**Comment by vshankar** [comment]
> > @lxbsz could you fix the yaml as done for quincy backport?
> 
> @vshankar Already done last Friday!

Ah, thx! didn't have the page refreshed!

**Review by vshankar — APPROVED** [review]

**Comment by yuriw** [comment]
@lxbsz @vshankar till can't schedule the `fs` suite

`teuthology-suite -v --ceph-repo $CEPH_REPO --suite-repo $CEPH_REPO -c $CEPH_BRANCH -m $MACHINE_NAME -s fs -k $DISTRO -p $PRIO -e $CEPH_QA_MAIL --suite-branch $CEPH_BRANCH -l $LIMIT -S $SHA1 --force-priority --subset 111/120000;
/home/yuriw/teuthology/virtualenv/lib/python3.6/site-packages/paramiko/transport.py:33: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.
  from cryptography.hazmat.backends import default_backend
2022-09-20 20:42:55,188.188 INFO:teuthology.suite:Using random seed=4136
2022-09-20 20:42:55,189.189 INFO:teuthology.suite.run:kernel sha1: distro
2022-09-20 20:42:55,421.421 INFO:teuthology.suite.run:ceph sha1 explicitly supplied
2022-09-20 20:42:55,422.422 INFO:teuthology.suite.run:ceph sha1: 054ae4f9059a14963ae5594a2b3b9c126e7ee054
2022-09-20 20:42:55,422.422 DEBUG:teuthology.suite.util:Defaults for machine_type smithi distro centos: arch=x86_64, release=centos/7, pkg_type=rpm
2022-09-20 20:42:55,422.422 INFO:teuthology.suite.util:container build centos/8, checking for build_complete
2022-09-20 20:42:55,422.422 DEBUG:teuthology.packaging:Querying https://shaman.ceph.com/api/search?status=ready&project=ceph&flavor=default&distros=centos%2F8%2Fx86_64&sha1=054ae4f9059a14963ae5594a2b3b9c126e7ee054
2022-09-20 20:42:56,033.033 DEBUG:teuthology.packaging:looking for centos/8 x86_64 default
2022-09-20 20:42:56,033.033 DEBUG:teuthology.packaging:build: centos/8 arm64 default
2022-09-20 20:42:56,033.033 DEBUG:teuthology.packaging:build: ubuntu/18.04 arm64 default
2022-09-20 20:42:56,034.034 DEBUG:teuthology.packaging:build: ubuntu/20.04 x86_64 default
2022-09-20 20:42:56,034.034 DEBUG:teuthology.packaging:build: centos/8 x86_64 default
2022-09-20 20:42:56,034.034 INFO:teuthology.suite.run:ceph version: 16.2.10-901.g054ae4f9
2022-09-20 20:42:56,270.270 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph-ci.git wip-yuri5-testing-2022-09-19-1007-pacific -> 054ae4f9059a14963ae5594a2b3b9c126e7ee054
2022-09-20 20:42:56,483.483 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/ceph-ci.git wip-yuri5-testing-2022-09-19-1007-pacific -> 054ae4f9059a14963ae5594a2b3b9c126e7ee054
2022-09-20 20:42:56,484.484 INFO:teuthology.suite.run:ceph-ci branch: wip-yuri5-testing-2022-09-19-1007-pacific 054ae4f9059a14963ae5594a2b3b9c126e7ee054
2022-09-20 20:42:56,494.494 DEBUG:teuthology.repo_utils:Setting repo remote to https://github.com/ceph/ceph-ci.git
2022-09-20 20:42:56,510.510 INFO:teuthology.repo_utils:Fetching wip-yuri5-testing-2022-09-19-1007-pacific from origin
2022-09-20 20:42:56,947.947 INFO:teuthology.repo_utils:Resetting repo at /home/yuriw/src/github.com_ceph_ceph-c_wip-yuri5-testing-2022-09-19-1007-pacific to origin/wip-yuri5-testing-2022-09-19-1007-pacific
2022-09-20 20:42:57,835.835 DEBUG:teuthology.suite.run:Check file /home/yuriw/src/github.com_ceph_ceph-c_wip-yuri5-testing-2022-09-19-1007-pacific/qa/.teuthology_branch exists
2022-09-20 20:42:58,042.042 DEBUG:teuthology.repo_utils:git ls-remote https://github.com/ceph/teuthology main -> b58f2c18636eb10faa77ed3614abd00cb85dfc2c
2022-09-20 20:42:58,042.042 INFO:teuthology.suite.run:teuthology branch: main b58f2c18636eb10faa77ed3614abd00cb85dfc2c
2022-09-20 20:42:58,056.056 DEBUG:teuthology.suite.run:Suite fs in /home/yuriw/src/github.com_ceph_ceph-c_wip-yuri5-testing-2022-09-19-1007-pacific/qa/suites/fs
2022-09-20 20:42:58,056.056 DEBUG:teuthology.suite.run:subset = (111, 120000)
2022-09-20 20:42:58,057.057 DEBUG:teuthology.suite.run:no_nested_subset = False
2022-09-20 20:42:58,057.057 INFO:teuthology.suite.build_matrix:Subset=111/120000
Traceback (most recent call last):
  File "/home/yuriw/teuthology/virtualenv/bin/teuthology-suite", line 33, in <module>
    sys.exit(load_entry_point('teuthology', 'console_scripts', 'teuthology-suite')())
  File "/home/yuriw/teuthology/scripts/suite.py", line 219, in main
    return teuthology.suite.main(args)
  File "/home/yuriw/teuthology/teuthology/suite/__init__.py", line 145, in main
    run.prepare_and_schedule()
  File "/home/yuriw/teuthology/teuthology/suite/run.py", line 422, in prepare_and_schedule
    num_jobs = self.schedule_suite()
  File "/home/yuriw/teuthology/teuthology/suite/run.py", line 589, in schedule_suite
    seed=self.args.seed)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 62, in build_matrix
    mat, first, matlimit = _get_matrix(path, subset, no_nested_subset)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 69, in _get_matrix
    mat = _build_matrix(path, mincyclicity=divisions, no_nested_subset=no_nested_subset)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 156, in _build_matrix
    fn)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 156, in _build_matrix
    fn)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 156, in _build_matrix
    fn)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 136, in _build_matrix
    fn)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 100, in _build_matrix
    fn)
  File "/home/yuriw/teuthology/teuthology/suite/build_matrix.py", line 80, in _build_matrix
    raise IOError('%s does not exist (abs %s)' % (path, os.path.abspath(path)))
OSError: /home/yuriw/src/github.com_ceph_ceph-c_wip-yuri5-testing-2022-09-19-1007-pacific/qa/suites/fs/upgrade/volumes/import-legacy/overrides/whitelist_health.yaml does not exist (abs /home/yuriw/src/github.com_ceph_ceph-c_wip-yuri5-testing-2022-09-19-1007-pacific/qa/suites/fs/upgrade/volumes/import-legacy/overrides/whitelist_health.yaml)
`

**Comment by vshankar** [comment]
> @lxbsz @vshankar till can't schedule the `fs` suite

Xiubo is on PTO. I'll try to look into the failure sometime this week.

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
@lxbsz Mind having a look at this - https://github.com/ceph/ceph/pull/47891#issuecomment-1252892080

(I couldn't get time to debug this when you were on PTO)

**Comment by lxbsz** [comment]
> @lxbsz Mind having a look at this - [#47891 (comment)](https://github.com/ceph/ceph/pull/47891#issuecomment-1252892080)
> 
> (I couldn't get time to debug this when you were on PTO)

Sorry I missed this PR.

Fixed it in https://github.com/ceph/ceph/pull/47891/commits/caa85f7871c787d72254bec03541addb4f3dff64 by:

```
create qa/suites/fs/upgrade/volumes/import-legacy/overrides/ignorelist_health.yaml
create qa/suites/fs/upgrade/volumes/import-legacy/overrides/ignorelist_wrongly_marked_down.yaml
delete qa/suites/fs/upgrade/volumes/import-legacy/overrides/whitelist_health.yaml
delete qa/suites/fs/upgrade/volumes/import-legacy/overrides/whitelist_wrongly_marked_down.yaml
```

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test api

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test api

File: qa/tasks/cephfs/cephfs_test_case.py [hunk 1]
```diff
@@ -90,6 +90,27 @@ def _save_mount_details(self):
         self._orig_mount_details = [MountDetails(m) for m in self.mounts]
         log.info(self._orig_mount_details)
 
+    def _remove_blocklist(self):
+        # In case anything is in the OSD blocklist list, clear it out.  This is to avoid
+        # the OSD map changing in the background (due to blocklist expiry) while tests run.
+        try:
+            self.mds_cluster.mon_manager.run_cluster_cmd("osd", "blocklist", "clear")
```
Comment (batrick) [inline qa/tasks/cephfs/cephfs_test_case.py:97]: Causes:

```
2022-11-30T21:54:56.844 INFO:tasks.cephfs_test_runner:======================================================================
2022-11-30T21:54:56.845 INFO:tasks.cephfs_test_runner:ERROR: test_inject_to_empty (tasks.cephfs.test_journal_repair.TestJournalRepair)
2022-11-30T21:54:56.845 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2022-11-30T21:54:56.846 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2022-11-30T21:54:56.846 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_8b53f015ec2658258282b5e71390dde213b67dbc/qa/tasks/cephfs/cephfs_test_case.py", line 153, in setUp
2022-11-30T21:54:56.847 INFO:tasks.cephfs_test_runner:    self._remove_blocklist()
2022-11-30T21:54:56.847 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/github.com_ceph_ceph-c_8b53f015ec2658258282b5e71390dde213b67dbc/qa/tasks/cephfs/cephfs_test_case.py", line 97, in _remove_blocklist
2022-11-30T21:54:56.848 INFO:tasks.cephfs_test_runner:    self.mds_cluster.mon_manager.run_cluster_cmd("osd", "blocklist", "clear")
2022-11-30T21:54:56.848 INFO:tasks.cephfs_test_runner:TypeError: run_cluster_cmd() takes 1 positional argument but 4 were given
```

/ceph/teuthology-archive/yuriw-2022-11-30_17:19:18-fs-wip-yuri2-testing-2022-11-30-0724-pacific-distro-default-smithi/7098613/teuthology.log

(and many others)

That method's signature changed in quincy or main.

**Review by batrick — CHANGES_REQUESTED** [review]

File: qa/tasks/cephfs/cephfs_test_case.py (hunk 1 above)
Comment (lxbsz) [reply qa/tasks/cephfs/cephfs_test_case.py:97]: I think it should be `raw_cluster_cmd()` instead here.

Comment (lxbsz) [reply qa/tasks/cephfs/cephfs_test_case.py]: Fixed it.

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Pacific#2022-Dec-07

**Comment by yuriw** [comment]
jenkins test this please

**Comment by yuriw** [comment]
@batrick pls merge is your comments addressed

@vshankar approved this

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test make check

**Comment by yuriw** [comment]
jenkins test make check

**Comment by vshankar** [comment]
make check fails with

```
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/ceph_fuse.py", line 9, in <module>
    from tasks.cephfs.fuse_mount import FuseMount
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/cephfs/fuse_mount.py", line 13, in <module>
    from tasks.cephfs.mount import CephFSMount
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/cephfs/mount.py", line 12, in <module>
    from IPy import IP
ModuleNotFoundError: No module named 'IPy'
```

**Comment by lxbsz** [comment]
> > make check fails with
> > ```
> >   File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/ceph_fuse.py", line 9, in <module>
> >     from tasks.cephfs.fuse_mount import FuseMount
> >   File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/cephfs/fuse_mount.py", line 13, in <module>
> >     from tasks.cephfs.mount import CephFSMount
> >   File "/home/jenkins-build/build/workspace/ceph-pull-requests/qa/tasks/cephfs/mount.py", line 12, in <module>
> >     from IPy import IP
> > ModuleNotFoundError: No module named 'IPy'
> > ```
> 
> #49363 is intended to fix a lot of the make check issues since the tox version upgrade. All the failures I see in the make check run here
> 
> ```
> The following tests FAILED:
> 	  1 - run-tox-mgr (Failed)
> 	  7 - run-tox-cephadm (Failed)
> 	202 - run-tox-qa (Failed)
> 	207 - run-tox-jsonnet-check (Failed)
> 	209 - run-tox-alerts-lint (Failed)
> ```
> 
> are passing on that PR with the exception of run-tox-qa which is failing with the same `ModuleNotFoundError: No module named 'IPy'` which means that failure is unrelated to anything in this PR and also isn't fixed by the changes made against main to get things working with the new tox version. It looks like on main and quincy we no longer run this `import-tasks` test. We could maybe drop it in pacific as well, or we'll have to figure out the dependency issue and install IPy at some point in the process.

In this PR it won't touch anything related to the `IPy`, but the `run-tox-qa` will always check the whole file the patches touched.

**Comment by vshankar** [comment]
> are passing on that PR with the exception of run-tox-qa which is failing with the same `ModuleNotFoundError: No module named 'IPy'` which means that failure is unrelated to anything in this PR and also isn't fixed by the changes made against main to get things working with the new tox version. It looks like on main and quincy we no longer run this `import-tasks` test. We could maybe drop it in pacific as well, or we'll have to figure out the dependency issue and install IPy at some point in the process.

Right - its unrelated but its a required check to merge a change.

+1 for dropping `import-tasks` in pacific if its not really necessary (however, I'm unaware of those developments).

**Comment by adk3798** [comment]
> > are passing on that PR with the exception of run-tox-qa which is failing with the same `ModuleNotFoundError: No module named 'IPy'` which means that failure is unrelated to anything in this PR and also isn't fixed by the changes made against main to get things working with the new tox version. It looks like on main and quincy we no longer run this `import-tasks` test. We could maybe drop it in pacific as well, or we'll have to figure out the dependency issue and install IPy at some point in the process.
> 
> Right - its unrelated but its a required check to merge a change.
> 
> +1 for dropping `import-tasks` in pacific if its not really necessary (however, I'm unaware of those developments).

Kefu ended up recommending dropping it over the weekend as well, so going with that right now. Make check is passing on the backport now too https://github.com/ceph/ceph/pull/49363#discussion_r1045104744

**Comment by lxbsz** [comment]
jenkins test make check

**Comment by adk3798** [comment]
jenkins test make check

**Comment by adk3798** [comment]
```
99% tests passed, 1 tests failed out of 211

Total Test time (real) = 680.56 sec

The following tests FAILED:
	 14 - run-rbd-unit-tests-109.sh (Failed)
```

**Comment by adk3798** [comment]
jenkins test make check

**Review by batrick — APPROVED** [review]

**Pull request merged**